src/proposal/freeze_history.mligo

```diff
@@ -18,12 +18,10 @@ let stake_frozen_fh (amt, fh : nat * address_freeze_history): address_freeze_his
   let fh = sub_frozen_fh(amt, fh) in
   { fh with staked = fh.staked + amt }
 
-let unstake_frozen_fh (amt, fh : nat * address_freeze_history): address_freeze_history =
-  match is_nat(fh.staked - amt) with
-  | Some new_amt ->
-      // Adding to past_unstaked should be fine since as of now, the staked
-      // tokens have to be from past stages.
-      { fh with staked = new_amt; past_unstaked = fh.past_unstaked + amt }
+let unstake_frozen_fh (amt_to_unstake, amt_to_burn, fh : nat * nat * address_freeze_history): address_freeze_history =
+  match is_nat(fh.staked - (amt_to_unstake + amt_to_burn)) with
+  | Some new_staked_amt ->
+      { fh with staked = new_staked_amt; past_unstaked = fh.past_unstaked + amt_to_unstake }
```

Comment:
May be we should add a test for this.

Reply:
With the new `checkBalance` function, it already uses `freeze_history` to check the balance, which means it covers `unstake_frozen_fh` as well. https://github.com/tqtezos/baseDAO/pull/275/files#diff-d519aedaf6d7160f71739847f8cdfe0224fdee4cd758cd76e5d56df3e62215f3R86

Or maybe you have some other tests in mind, could suggest to me as well.

Reply:
I see. AFAIU, it now check the freeze balance by summing all the field in `address_freeze_history`, so cannot detect if two field values are swapped (because the sum will remain the same). May be it is good enough. I ll leave it up to you.